### PR TITLE
Add anchors to show/hide sections in install page

### PR DESCRIPTION
**Pull request by abbycross**
EDIT: Adding custom anchor tags to the show/hide sections in the "Qiskit versioning" section.

Once this is merged, direct links to the show-hide sections will be:

https://docs.quantum.ibm.com/start/install#release-schedule
https://docs.quantum.ibm.com/start/install#upgrade-strategy
https://docs.quantum.ibm.com/start/install#pre-releases
https://docs.quantum.ibm.com/start/install#post-releases

---------------------------
(Original proposal below, for historical purposes; we are no longer separating out the versioning section to a new page.)
In discussions with @mtreinish, the proposal is to separate out the versioning section (currently at the end of the start/install page) into a new page, for the following reasons:

- it really isn't directly install-related
- it allows us to remove the show/hide features for that content, which makes it easier to link to and find

We may also want to remove the show/hide features for the OS support tiers, but leave them in the current install page, since they are install-related. 

Topics to discuss:

- [ ] What to put in Next steps section? Currently using the same as what is on the Install pg as a placeholder.
- [ ] Title at the moment is just Qiskit versioning (since that's the only thing in the new page); are we good with this?
- [ ] I put the new page after the "Advanced setup" section in the toc. Should it be elsewhere?

cc: @mtreinish @Eric-Arellano

### CODE REVIEW

**Comment by abbycross** [comment]
Ignore my first commit -- it's no longer the direction we're going in!!

File: docs/start/qiskit-versioning.mdx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+If there are issues with the packaging of a given release, a post-release may be
+issued to correct this. These will follow the form `X.Y.Z.1` where the fourth
+integer is used to indicate it is the first post-release of the `X.Y.Z` release.
+For example, the qiskit-terra (the legacy package name for Qiskit) 0.25.2
+release had some issue with the sdist package publishing, and a post-release
+0.25.2.1 was published that corrected this issue. The code was identical, and
+0.25.2.1 only fixed the packaging issue for the release.
+
+
+## Next steps
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:128]: I think the only topics that apply as next steps would be Install and set up Qiskit, and Configure Qiskit locally.  And maybe Release notes (https://docs.quantum.ibm.com/api/qiskit/release-notes)

File: docs/start/qiskit-versioning.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+title: Qiskit versioning
+description: Description of how qiskit version numbers work, and information about release schedules, upgrade strategy, pre-releases, and post-releases
+
+---
+
+# Qiskit versioning
+
+Qiskit version numbers follow [Semantic Versioning](https://semver.org/).
+The version number is comprised of three primary components: the major, minor, and
+patch versions. For a version number `X.Y.Z` where `X` is the major version,
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:11]:
```suggestion
patch versions. For example, in version number `X.Y.Z`, `X` is the major version,
```

File: docs/start/qiskit-versioning.mdx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+[milestones list](https://github.com/Qiskit/qiskit/milestones), which will always contain the current release plan.
+
+With the release of a new major version, the previous major version is supported
+for at least six months; only bug and security fixes will be accepted during this time and only patch releases will be published for this major version. A final
+patch version will be published when support is dropped, and that release will
+also document the end of support for that major version series. A longer
+support window is needed for the previous major version as this gives downstream
+consumers of Qiskit a chance to migrate not only their code but also their
+users. We do not recommended for a downstream library that
+depends on Qiskit to bump its minimum Qiskit version to a new
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:40]:
```suggestion
depend on Qiskit should not raise their minimum required Qiskit version to a new
```

File: docs/start/qiskit-versioning.mdx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+major version release immediately because its user base will also need a chance
+to migrate to the new API changes. Having an extended support window
+for the previous major Qiskit version gives downstream projects time to fix
+compatibility with the next major version. Downstream projects can provide
+support for two release series at a time to give their users a migration path.
+
+For the purposes of semantic versioning, the Qiskit public API is considered
+any documented module, class, function, or method that is not marked as private
+(with a `_` prefix). However, there can be explicit exceptions made in the case
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:49]:
```suggestion
(with an underscore `_` prefix). However, there can be explicit exceptions made for
```

File: docs/start/qiskit-versioning.mdx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+of specific documented APIs. In such cases these APIs will be clearly documented
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:50]:
```suggestion
specific documented APIs. In such cases, these APIs will be clearly documented
```

File: docs/start/qiskit-versioning.mdx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+version releases will raise minimum requirements for using or building Qiskit
+(including adding new dependencies), but patch fixes might include support for
+new versions of Python or other dependencies. Usually the minimum version of a
+dependency is only increased when older dependency versions go out of support or
+when it is not possible to maintain compatibility with the latest release of the
+dependency and the older version.
+
+## Upgrade strategy
+
+Whenever a new major version is released, the recommended upgrade path
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:74]:
```suggestion
When a new major version is released, the recommended upgrade path
```

File: docs/start/qiskit-versioning.mdx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+is to first upgrade to use the most recent minor version on the previous major
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:75]:
```suggestion
is to first upgrade to the most recent minor version on the previous major
```

File: docs/start/qiskit-versioning.mdx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+version. Shortly before a new major version, a final minor version will
+be published. This final minor version release `M.N+1.0` is equivalent to
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:77]:
```suggestion
be published. This final minor version release `X.Y+1.0.0` is equivalent to
```

File: docs/start/qiskit-versioning.mdx [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+`M.N.0` but with warnings and deprecations for any API changes that are
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:78]:
```suggestion
`X.Y.0` but with warnings and deprecations for any API changes that are
```

File: docs/start/qiskit-versioning.mdx [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+made on the new major version series.
+
+For example, immediately proceeding the 1.0.0 release, a 0.46.0 release will be
+published. The 0.46.0 release will be equivalent to the 0.45.0 release but with
+additional deprecation warnings that document the API changes that were made as
+part of the 1.0.0 release. This pattern will be used for any future major
+version releases.
+
+As a user of Qiskit, it's recommended that you first upgrade to this final minor
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:87]:
```suggestion
Qiskit users should first upgrade to this final minor
```

File: docs/start/qiskit-versioning.mdx [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+version first, so you can see any deprecation warnings and adjust your Qiskit
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:88]:
```suggestion
version to see any deprecation warnings and adjust their Qiskit
```

File: docs/start/qiskit-versioning.mdx [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+usage ahead of time before trying a potentially breaking release. The previous
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:89]:
```suggestion
usage before trying a potentially breaking release. The previous
```

File: docs/start/qiskit-versioning.mdx [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+major version will be supported for at least six months to give sufficient time
+to upgrade. A typical pattern to manage this is to pin the max version to
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:91]:
```suggestion
to upgrade. A typical pattern to manage this is to pin the maximum version to
```

File: docs/start/qiskit-versioning.mdx [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+avoid using the next major release series until you're sure of compatibility.
+For example, specifying in a requirements file `qiskit<2` when the current
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:93]:
```suggestion
For example, specifying `qiskit<2` in a requirements file when the current
```

File: docs/start/qiskit-versioning.mdx [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+major Qiskit version is 1 will ensure that you're using a version of Qiskit
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:94]:
```suggestion
major Qiskit version is 1 ensures that you're using a version of Qiskit
```

File: docs/start/qiskit-versioning.mdx [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+that won't have breaking API changes.
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:95]:
```suggestion
that doesn't have breaking API changes.
```

File: docs/start/qiskit-versioning.mdx [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+
+Pre-emptively capping the version less than the next major version is necessary
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:97]:
```suggestion
Capping the version less than the next major version 
```

File: docs/start/qiskit-versioning.mdx [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+to ensure you get a chance to see deprecation warnings before a
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:98]:
```suggestion
ensures that you see any deprecation warnings before a
```

File: docs/start/qiskit-versioning.mdx [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+major version release. The normal release schedule means the last minor
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:99]:
```suggestion
major version release. 
```

File: docs/start/qiskit-versioning.mdx [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+version release which includes any final deprecation warnings will be released
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:100]:
```suggestion
```

File: docs/start/qiskit-versioning.mdx [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+shortly before the next major version and `pip` will default to using
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:101]:
```suggestion
Without the cap, `pip` installs 
```

File: docs/start/qiskit-versioning.mdx [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+the newest version available unless the version cap is set.
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:102]:
```suggestion
the newest version available by default.
```

File: docs/start/qiskit-versioning.mdx [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
+
+## Pre-releases
+
+For each minor and major version release Qiskit will publish pre-releases that
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:106]:
```suggestion
For each minor and major version release Qiskit publishes pre-releases that
```

File: docs/start/qiskit-versioning.mdx [hunk 24]
```diff
@@ -0,0 +1,135 @@
...
+
+Note that when one of the PEP440 pre-release suffixes (such as `a`, `b`, or `pre`) are
+published, it does not have the same guarantees as an `rc` release, and is
+only a preview release. The API might change between these pre-releases
+and the final release with that version number. For example, `1.0.0pre1` might have
+a different final API than `1.0.0`.
+
+## Post-releases
+
+If there are issues with the packaging of a given release, a post-release may be
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:119]:
```suggestion
If there are issues with a release's packaging, a post-release might be
```

File: docs/start/qiskit-versioning.mdx [hunk 25]
```diff
@@ -0,0 +1,135 @@
...
+issued to correct this. These will follow the form `X.Y.Z.1` where the fourth
+integer is used to indicate it is the first post-release of the `X.Y.Z` release.
```
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:121]:
```suggestion
integer indicates that it is the first post-release of the `X.Y.Z` release.
```

**Review by beckykd — APPROVED** [review]

File: docs/start/qiskit-versioning.mdx [hunk 26]
```diff
@@ -0,0 +1,134 @@
...
+0.25.2.1 only fixed the packaging issue for the release.
+
+
+## Next steps
+
+<Admonition type="tip" title="Recommendations">
+  -  [Select and set up an IBM Quantum channel.](setup-channel)
+  -  [Configure Qiskit locally.](configure-qiskit-local)
+  -  Follow the steps in [Hello world](hello-world) to write and run a quantum program.
+  -  Try one of the [workflow example tutorials.](https://learning.quantum.ibm.com/catalog/tutorials?category=workflow-example)
```
Comment (abbycross) [inline docs/start/qiskit-versioning.mdx:133]:
```suggestion
  -  [Install and set up Qiskit.](install)
  -  [Configure Qiskit locally.](configure-qiskit-local)
  -  Visit the Qiskit [release notes.](/api/qiskit/release-notes) 
```

File: docs/start/install.mdx [hunk 27]
```diff
@@ -147,181 +147,35 @@ Qiskit strives to support as many operating systems as possible, but due to limi
...
-  0.25.2.1 was published that corrected this issue. The code was identical, and
-  0.25.2.1 only fixed the packaging issue for the release.
-</details>
+- Linux ppc64le (distributions compatible with the [manylinux 2014](https://www.python.org/dev/peps/pep-0599/)packaging specification)
+- Linux s390x (distributions compatible with the [manylinux 2014](https://www.python.org/dev/peps/pep-0599/) packaging specification)
+- macOS ARM64 (10.15 or newer)
+- Linux i686 (distributions compatible with the [manylinux 2014](https://www.python.org/dev/peps/pep-0599/) packaging specification)
+- Windows 32 bit
 
 ## Next steps
```
Comment (Eric-Arellano) [inline docs/start/install.mdx:181]: Maybe link to the new page in Next Steps.

File: docs/start/qiskit-versioning.mdx [hunk 28]
```diff
@@ -0,0 +1,133 @@
+---
+title: Qiskit versioning
+description: Description of how qiskit version numbers work, and information about release schedules, upgrade strategy, pre-releases, and post-releases
```
Comment (Eric-Arellano) [inline docs/start/qiskit-versioning.mdx:3]: Great description!

```suggestion
description: Description of how Qiskit version numbers work, and information about release schedules, upgrade strategy, pre-releases, and post-releases.
```

File: docs/start/qiskit-versioning.mdx [hunk 29]
```diff
@@ -0,0 +1,133 @@
...
+the most recent minor version of each actively supported release series (that is, the 
+major version). We support at most two release series at a time, which occurs
+only during the period of overlap following a new major version release,
+described in more detail below.
+
+## Release schedule
+
+A tentative release schedule is included below:
+
+![](/images/start/qiskit-versioning/release_schedule.png)
```
Comment (Eric-Arellano) [inline docs/start/qiskit-versioning.mdx:28]: Thanks for following the image file format <3

File: docs/start/_toc.json [hunk 30]
```diff
@@ -35,6 +35,10 @@
           "url": "/start/configure-qiskit-local"
         }
       ]
+    },
+    {
+      "title": "Qiskit versioning",
+      "url": "/start/qiskit-versioning"
     }
```
Comment (Eric-Arellano) [inline docs/start/_toc.json:42]: I think this makes sense as top-level, rather than "Advanced setup". But I'm curious what others think.

**Review by Eric-Arellano — APPROVED** [review]
Looks great to me! Maybe wait for sign-off from Abby M

File: docs/start/_toc.json (hunk 30 above)
Comment (javabster) [reply docs/start/_toc.json:42]: I'm not super keen on having it be a top level  page in the start section. We're really trying to minimise the amount of content there. what about moving this page back to the api reference? I remember we briefly discussed that last year, what do you think?

Comment (Eric-Arellano) [reply docs/start/_toc.json:42]: Where would that go in `/api`? A new section like `/api/open-source`?

**Comment by abbycross** [comment]
Update: @javabster suggested trying 

> `<summary id="anchor-name">Description words</summary>` 

within the `<details>` tags, and this works much better than `<span>`. In theory we can bring the show/hide sections back and keep everything on the same page -- however, we still do not have a way for the reader to copy an anchor's link (no little chain icon shows up). That alone may be reason enough to split out the content into a separate page; otherwise the reader would have to look at the code to discover the anchor tag. 

One option is to include a header within the show/hide content itself, which would show the chain icon, AND this means the header would show up on the right nav (if it's h2 or h3), but then we would have a duplicated line situation like this:
Screenshot 2024-01-10 at 2 51 29 PM



Another option is to create the header right within the summary tags, which would work great if the chain icon wasn't superimposed with the caret to click on!!
Screenshot 2024-01-10 at 2 47 01 PM

Thoughts or reasonable workarounds appreciated!... I have scoured the internet but perhaps am not using the right search terms...

**Comment by abbycross** [comment]
Had an offline conversation with @javabster and we determined that the best way to move forward is to return to the original formatting of this content and create custom anchor links for the show/hide sections. In the future we hope to have a better component for the show-hide sections so that we can more flexibly link to them.

**Comment by abbycross** [comment]
One last comment: while this functionality works, the links may appear only to jump to the Qiskit versioning section, because the show/hide section is so close to the end of the page. 🤷

**Pull request merged**